Week7/REVIEW.md

```diff
@@ -1,9 +1,196 @@
 # REVIEW JavaScript week 7
 
+## Git Workflow
+
+To be provided
+
+## Map, filter, reduce 
+
+The array methods **map()**, **filter()** and **reduce()** are best understood by looking at how they could be implemented if we were to write them ourselves. In the next few sections we will present simplified versions of the native implementations. We have prefixed the method names with `my` to distinguish them from the built-in versions.
+
+### Array#map\*
+
+The **map** method returns a new array where each element of the subject array is transformed by a user-supplied transformation (= _mapping_) function. 
+
+```js
+Array.prototype.myMap = function (mapFn) {
+  const arr = [];
+  for (let i = 0; i < this.length; i++) {
+    arr.push(mapFn(this[i], i, this));
+  }
+  return arr;
+};
+```
+
+<small>\* Array#map is a short-hand notation for Array.prototype.map.</small>
+
+As you can see, the `mapFn` function is called with three arguments:
+
+1. the current array element to be transformed
+2. the index of the element (starting with `0`)
+3. the subject array itself
+
+As is usual in JavaScript you do not necessarily have to use all the parameters that were passed to the `mapFn` function. In fact, in many cases you will only need the first argument (the current array element).
+
+In the example below we will use the Array#map method to compute the squares of an array of numbers.
+
+```js
+const numbers = [3, 5, 2, 7];
+const squares = numbers.map(num => num * num);
+console.log(squares); // -> [9, 25, 4, 49]
+```
+
+### Array#filter
+
+The **filter()** method returns a new array with all elements that pass the test implemented by a user-supplied (predicate\*) function. 
+
```

Comment:
maybe you can describe a bit more what the function below does, (in a simpler way)